js/modules/loki_snode_api.js

```diff
@@ -13,30 +16,119 @@ const snodeHttpsAgent = new https.Agent({
 
 const RANDOM_SNODES_TO_USE_FOR_PUBKEY_SWARM = 3;
 const SEED_NODE_RETRIES = 3;
+const SNODE_VERSION_RETRIES = 3;
+
+// findMatchingSnode(search)(current)
+const findMatchingSnode = search => current =>
```

Comment:
Not sure this the name describes well what the function is actually doing (it is not doing any searching). How about `matchesSnode`? Also, should we just use `pubkey_ed25519` instead to compare nodes?

Reply:
makeMatchesSnodeFunc?
All unified now, so you can change it to how ever you want to search

Reply:
I like `snodes.some(matchesSnode(item));` better than `snodes.some(makeMatchesSnodeFunc(item));`.

Reply:
but it returns a function and i think that's important to denote in the name